src/lib/injector_preload/injector.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * The Shadow Simulator
+ * See LICENSE for licensing information
+ */
+
+#include <stddef.h>
+#include <syscall.h>
+
+#include "lib/shim/shim_api.h"
+
+/// The purpose of the injector library is to inject the shim into the process
+/// space of each managed process, since the shim is needed for interaction with
+/// Shadow. (Technically, the shim will already be injected if there are other
+/// preloaded libraries that link to it, but this library provides a minimally
+/// invasive way to inject the shim without preloading other symbols.)
+///
+/// One goal is to ensure the shim is injected, so we link to it and call one
+/// shim function in a constructor. The other goal is to be minimally invasive,
+/// so let's not define any other symbols here.
```

Comment:
Might be useful to add a comment here that we preload the injector lib rather than the preload the shim directly so that we inject the shim without intercepting any functions from the managed process.

Reply:
I updated and expanded the comment.